sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/InMemoryCatalog.scala

```diff
@@ -343,7 +343,9 @@ class InMemoryCatalog(
 
   override def getTable(db: String, table: String): CatalogTable = synchronized {
     requireTableExists(db, table)
-    catalog(db).tables(table).table
+    val catalogTable = catalog(db).tables(table).table
+    val catalogOption = CatalystIdentifier.sessionCatalogOption(catalogTable.identifier.database)
+    catalogTable.copy(identifier = catalogTable.identifier.copy(catalog = catalogOption))
```

Comment:
how about we have a `CatalystIdentifier.attachCatalog`?
```
catalogTable.copy(identifier = CatalystIdentifier.attachCatalog(atalogTable.identifier))
```

Reply:
The benefit is we can make the code safer: we should not overwrite the catalog field if it's already specified.

Reply:
added attachCatalog method in CatalystIdentifier